kpatch-build/kpatch-build

```diff
@@ -434,76 +434,121 @@ find_special_section_data() {
 	return
 }
 
-filter_parent_obj()
-{
-  local dir="${1}"
-  local file="${2}"
-
-  grep -v "\.mod\.cmd$" | grep -Fv "${dir}/.${file}.cmd"
-}
+find_parent_obj_in_dir() {
+	local file="$1"
+	local dir="$2"
+
+	local filedir
+	local filename
+	local parent_dir
+	declare -a parents
+
+	# make sure the dir has .cmd files
+	if ! compgen -G "$dir"/.*.cmd > /dev/null; then
+		PARENT=""
+		return
+	fi
 
-find_parent_obj() {
-	dir="$(dirname "$1")"
-	absdir="$(readlink -f "$dir")"
-	pwddir="$(readlink -f .)"
-	pdir="${absdir#$pwddir/}"
-	file="$(basename "$1")"
-	grepname="${1%.o}"
-	grepname="$grepname\\.o"
-	if [[ "$DEEP_FIND" -eq 1 ]]; then
-		num=0
-		if [[ -n "$last_deep_find" ]]; then
-			parent="$(grep -lw "$grepname" "$last_deep_find"/.*.cmd | filter_parent_obj "${pdir}" "${file}" | head -n1)"
-			num="$(grep -lw "$grepname" "$last_deep_find"/.*.cmd | filter_parent_obj "${pdir}" "${file}" | wc -l)"
-		fi
-		if [[ "$num" -eq 0 ]]; then
-			parent="$(find . -name ".*.cmd" -print0 | xargs -0 grep -lw "$grepname" | filter_parent_obj "${pdir}" "${file}" | cut -c3- | head -n1)"
-			num="$(find . -name ".*.cmd" -print0 | xargs -0 grep -lw "$grepname" | filter_parent_obj "${pdir}" "${file}" | wc -l)"
-			[[ "$num" -eq 1 ]] && last_deep_find="$(dirname "$parent")"
-		fi
-	else
-		parent="$(grep -lw "$grepname" "$dir"/.*.cmd | filter_parent_obj "${dir}" "${file}" | head -n1)"
-		num="$(grep -lw "$grepname" "$dir"/.*.cmd | filter_parent_obj "${dir}" "${file}" | wc -l)"
+	filedir="$(dirname "$file")"
+	filename="$(basename "$file")"
+
+	# Linux < 5.19 file path is relative to kernel source tree root
+	while IFS='' read -r line; do
+		parents+=("$line")
+	done < <(grep -Flw "$file" "$dir"/.*.cmd | grep -Fv "$filedir/.$filename.cmd")
```

Comment:
Hello,
Compared to previous filter_parent_obj, should this line be as follows?
```
         done < <(grep -Flw "$file" "$dir"/.*.cmd | grep -v "\.mod\.cmd$" | grep -Fv "$filedir/.$filename.cmd")
```
At least this fixes the "two parent matches" build error of macro-callback integration test on kernel 5.18 with using this new version. (I have not tested fully nor 5.19 kernel yet)

Reply:
Also, "ERROR: two parent matches for fs/nfsd/export.o" from module.patch on rhel-9.